common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbDate.java

```diff
@@ -28,58 +30,131 @@
 import java.time.chrono.IsoChronology;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
 import java.time.temporal.ChronoUnit;
 import java.time.temporal.TemporalAccessor;
-import java.util.Date;
-import java.util.GregorianCalendar;
+import java.util.Arrays;
 import java.util.Locale;
 import java.util.function.BiFunction;
 
-public class TbDate extends Date {
+public class TbDate implements Serializable, Cloneable {
 
-    private static final DateTimeFormatter isoDateFormatter = DateTimeFormatter.ofPattern(
-            "yyyy-MM-dd[[ ]['T']HH:mm[:ss[.SSS]][ ][XXX][Z][z][VV][O]]").withZone(ZoneId.systemDefault());
+    private Instant instant;
+    private ZonedDateTime zonedDateTime;
```

Comment:
You don't need these variables. Instead write two methods: 

getUtcDateTime();
getLocalDateTime();